docs(rfc): add declarative configuration rfc and rfc template (#328)
getoutreach/devbase: rfcs/0-template.md

> # Declarative Configuration

Author: ```suggestion
# Title of Proposal
```

<!-- Add more dates here post-merge -->

| Date       | Status   |
|------------|----------|
| YYYY-MM-DD | Draft    |

## Table of Contents

<!-- toc -->
- [Summary](#summary)
- [Motivation](#motivation)
  - [Goals](#goals)
  - [Non-Goals](#non-goals)
- [Implementation Details](#implementation-details)
- [Test Plan](#testing-and-release-plan)
<!-- /toc -->

## Summary

A high level summary, without going into to much technical details, about what this RFC is about.

## Motiviation

Detail why this proposal is needed, without talking about the goals/non-goals and implementation details. Ideally this is just what is causing you to want to do this work. What pain-points are you trying to solve?

### Goals

Add a check list of what the goals of this proposal is, ideally this will help clarify what this is meant to do.

### Non-goals

> Add a check list of things this proposal does _not_ intend to solve.

Reviewer: Why is this a checklist?

You should mirror the goals description wording as well:
```suggestion
Add a list of things this proposal is _not_ trying to achieve.
```

## Implementation Details

> Add implementation specific details, the "how" of the proposal.

Reviewer: ```suggestion
Add details of how the proposal will be implemented.
```
